lib/core.mli

```diff
@@ -280,6 +280,22 @@ val split : ?pos:int -> ?len:int -> re -> string -> string list
         # Re.split ~pos:3 regex "1,2,3,4. Commas go brrr.";;
         - : string list = ["3"; "4. Commas go brrr."]
     ]}
+
+    Be careful when using [split] with zero-length patterns like [eol], [bow], 
```

Comment:
Perhaps you should explain what to be careful about, as it's kind of vague even with the examples. I get the general idea, but I think this could be explained in more detail.

Reply:
I added more details in the latest commit.